File: Specialization_Data_Structures_Algorithms_UC_San_Diego/Course1_Algorithmic_Toolbox/week2_Introduction/lcm/lcm.cpp
#include <iostream>

long long lcm_naive(int a, int b) {
  for (long l = 1; l <= (long long)a * b; ++l)
    if (l % a == 0 && l % b == 0)
      return l;

  return (long long)a * b;
}

int gcd_fast(int a, int b) {
  if (b == 0)
    return a;
  return gcd_fast(b, a % b);
}

long long lcm_fast(int a, int b) { return a / gcd_fast(a, b) * b; }

int main() {
  int a, b;
  std::cin >> a >> b;
  std::cout << lcm_fast(a, b) << std::endl;
  return 0;
}

File: Specialization_Data_Structures_Algorithms_UC_San_Diego/Course1_Algorithmic_Toolbox/week2_Introduction/lcm/lcm.py
# Uses python3
import sys

def lcm_naive(a, b):
    for l in range(1, a*b + 1):
        if l % a == 0 and l % b == 0:
            return l

    return a*b

def gcd_fast(a, b):
    if b==0 :
        return a
    return gcd_fast(b, a%b)

def lcm_fast(a, b):
    return a//gcd_fast(a,b)*b

if __name__ == '__main__':
    input = sys.stdin.read()
    a, b = map(int, input.split())
    print(lcm_fast(a, b))

